readthedocs/builds/tests/test_tasks.py

```diff
@@ -124,6 +124,47 @@ def test_archive_builds(self, build_commands_storage):
         self.assertEqual(Build.objects.filter(cold_storage=True).count(), 5)
         self.assertEqual(BuildCommandResult.objects.count(), 50)
 
+    def _create_builds(self, count, success=False):
+        """Helper to create a series of builds."""
+        builds = []
+        for _ in range(count):
+            build = get(
+                Build,
+                project=self.project,
+                version=self.version,
+                success=success,
+                state=BUILD_STATE_FINISHED,
+            )
+            builds.append(build)
+        return builds
+
+    @override_settings(RTD_BUILDS_MAX_CONSECUTIVE_FAILURES=50)
+    def test_task_disables_project_at_max_consecutive_failed_builds(self):
+        """Test that the project is disabled at the failure threshold."""
+        project = get(Project, slug="test-project", n_consecutive_failed_builds=False)
+        version = project.versions.get(slug=LATEST)
+        version.active = True
+        version.save()
+
+        # Create failures at the threshold
+        self._create_builds(settings.RTD_BUILDS_MAX_CONSECUTIVE_FAILURES + 1, success=False)
```

Comment:
We should probably have at least one more test case, when the project has at least one successful build.